open_spiel/python/algorithms/best_response.py

```diff
@@ -124,26 +133,63 @@ def transitions(self, state):
       # Counterfactual reach probabilities exclude the best-responder's actions,
       # hence return probability 1.0 for every action.
       return [(action, 1.0) for action in state.legal_actions()]
-    elif state.is_chance_node():
+    if state.is_chance_node():
       return state.chance_outcomes()
-    else:
-      return list(self._policy.action_probabilities(state).items())
+    if state.is_simultaneous_node():
+      action_prob_dicts = []
```

Comment:
This block looks rather complex to be in-line. Coud it be extracted into one or more helper functions (which could be independently tested)?